# fix:
#frontend.cpp: In function 'void startCampaignSelector()':
#../lib/framework/string_ext.h:178:71: error: format not a string literal and no format arguments [-Werror=format-security]
# #define ssprintf(dest, ...) snprintf((dest), sizeof(dest), __VA_ARGS__)
#                                                                       ^
#frontend.cpp:381:3: note: in expansion of macro 'ssprintf'
#   ssprintf(hackList[i], list[i].name.toUtf8().constData()); // since widget system is crazy and takes pointers not copies
#   ^
#../lib/framework/string_ext.h:178:71: error: format not a string literal and no format arguments [-Werror=format-security]
# #define ssprintf(dest, ...) snprintf((dest), sizeof(dest), __VA_ARGS__)
#                                                                       ^
#frontend.cpp:381:3: note: in expansion of macro 'ssprintf'
#   ssprintf(hackList[i], list[i].name.toUtf8().constData()); // since widget system is crazy and takes pointers not copies
#
# also fix
#../lib/sdl/libsdl.a(main_sdl.o): In function `inputhandleText(SDL_TextInputEvent*)':
#lib/sdl/main_sdl.cpp:1108: undefined reference to `UTF8toUTF32(char const*, unsigned int*)'
Index: warzone2100-3.2.0/src/frontend.cpp
===================================================================
--- warzone2100-3.2.0.orig/src/frontend.cpp	2016-07-12 20:29:08.997230958 +0200
+++ warzone2100-3.2.0/src/frontend.cpp	2016-07-12 20:29:08.993230848 +0200
@@ -378,7 +378,7 @@
 	QList<CAMPAIGN_FILE> list = readCampaignFiles();
 	for (int i = 0; i < list.size(); i++)
 	{
-		ssprintf(hackList[i], list[i].name.toUtf8().constData()); // since widget system is crazy and takes pointers not copies
+		ssprintf(hackList[i], "%s", list[i].name.toUtf8().constData()); // since widget system is crazy and takes pointers not copies
 		addTextButton(FRONTEND_CAMPAIGN_1 + i,  FRONTEND_POS1X, FRONTEND_POS2Y + 40 * i, hackList[i], WBUT_TXTCENTRE);
 	}
 	addSideText(FRONTEND_SIDETEXT, FRONTEND_SIDEX, FRONTEND_SIDEY, _("CAMPAIGNS"));
Index: warzone2100-3.2.0/tests/Makefile.am
===================================================================
--- warzone2100-3.2.0.orig/tests/Makefile.am	2016-07-12 20:29:08.953229746 +0200
+++ warzone2100-3.2.0/tests/Makefile.am	2016-07-12 23:25:01.490383990 +0200
@@ -21,8 +21,8 @@
 framework_linktest_LDADD = $(top_builddir)/lib/framework/libframework.a $(PHYSFS_LIBS) $(LIBCRYPTO_LIBS) $(LDFLAGS)
 
 ivis_linktest_SOURCES = ivis_linktest.cpp
-ivis_linktest_LDADD = $(top_builddir)/lib/framework/libframework.a $(top_builddir)/lib/ivis_opengl/libivis_opengl.a \
-	$(top_builddir)/lib/sdl/libsdl.a \
+ivis_linktest_LDADD = $(top_builddir)/lib/sdl/libsdl.a $(top_builddir)/lib/framework/libframework.a \
+	$(top_builddir)/lib/ivis_opengl/libivis_opengl.a \
 	$(PHYSFS_LIBS) $(LIBCRYPTO_LIBS) $(QT5_LIBS) $(SDL_LIBS) $(OPENGL_LIBS) $(OPENGLC_LIBS) $(GLEW_LIBS) $(QUESOGLC_LIBS) \
 	$(X_LIBS) $(X_EXTRA_LIBS) $(LDFLAGS) $(PNG_LIBS)
 
